### PR TITLE
feat(integrations): Add in feature flag check to Vercel Webhooks

### PR DESCRIPTION
**Pull request by roggenkemper**
this pr adds in a check so that we can turn on the modern Vercel Webhooks and slowly roll it out. The flow now will be both the modern/legacy requests are sent, but only one of them will run, depending on if you have the feature flag or not. So only one of the requests is actually processed.

### CODE REVIEW

File: src/sentry/integrations/vercel/webhook.py [hunk 1]
```diff
@@ -255,7 +260,10 @@ def _delete(self, external_id, configuration_id, request):
         #      organization integration AND the integration (since there is only one)
         #  2.) we already deleted the organization integration tied to this configuration
         #      and the remaining one is for a different org (and configuration)
-        if len(orgs) == 1:
+        if (
+            len(orgs) == 1
+            and features.has("organizations:vercel-integration-webhooks", orgs[0]) == new_webhook
```
Comment (wedamija) [inline src/sentry/integrations/vercel/webhook.py:265]: I just want to confirm that this is the expected logic here:
 - feature enabled: we end up in this branch when new_webhook is true, and not when it's false
 - feature disabled: we end up in this branch when new_webhook is False, and not when True

File: src/sentry/integrations/vercel/webhook.py [hunk 2]
```diff
@@ -191,22 +191,24 @@ def post(self, request: Request) -> Response | None:
...
                     configuration_id = payload["configuration"]["id"]
-                    return self._delete(external_id, configuration_id, request)
+                    return self._delete(external_id, configuration_id, request, new_webhook)
                 if event_type == "deployment.created":
-                    return self._deployment_created(external_id, request)
+                    return self._deployment_created(external_id, request, new_webhook)
             except Exception:
                 external_id = self.parse_old_external_id(request)
                 scope.set_tag("vercel_webhook.type", "old")
+                new_webhook = False
```
Comment (wedamija) [inline src/sentry/integrations/vercel/webhook.py:204]: Is there any other way to check for this other than relying on an exception here? I'm a little confused about how this branch works, at what point is the exception thrown?

Comment (roggenkemper) [reply src/sentry/integrations/vercel/webhook.py:204]: I'm pretty sure the Exception is coming from `parse_new_eternal_id`. the new payload has `team` while the legacy version has `teamId` (or user/userId) so if it unable to find that key, it throws the Exception and it tries the old method

Comment (wedamija) [reply src/sentry/integrations/vercel/webhook.py:204]: Ok, I think we should rewrite this to be a little clearer. The code in these branches is pretty similar. Something roughly like
```
try:
    external_id = self.parse_new_external_id(request)
    new_webhook = True
    tag_type_string = 'new'
except <Better exception than Exception, we should be able to be specific. AttributeError?>:
    external_id = self.parse_old_external_id(request)
    new_webhook = False
    tag_type_string = 'old'

scope.set_tag("vercel_webhook.type", tag_type_string)

if event_type == "integration-configuration.removed":
    configuration_id = request.data["payload"]["configuration"]["id"]
    return self._delete(external_id, configuration_id, request, new_webhook)
if event_type == "deployment.created":
    return self._deployment_created(external_id, request, new_webhook)
```

Comment (roggenkemper) [reply src/sentry/integrations/vercel/webhook.py:204]: agreed - thanks for the suggestion

File: src/sentry/integrations/vercel/webhook.py [hunk 3]
```diff
@@ -186,46 +186,45 @@ def post(self, request: Request) -> Response | None:
...
-                    return self._deployment_created(external_id, request)
-
+                new_webhook = False
+                tag_type_string = "old"
+
+            scope.set_tag("vercel_webhook.type", tag_type_string)
+
+            if (
+                event_type == "integration-configuration.removed"
+                or event_type == "integration-configuration-removed"
```
Comment (wedamija) [inline src/sentry/integrations/vercel/webhook.py:202]: Good catch, I didn't notice these were different.

Nit: `event_type in (...)`

**Review by wedamija — APPROVED** [review]

**Review by leeandher — APPROVED** [review]
looks good, much easier to read now 🚢

**Pull request merged**